quarkus/runtime/src/main/java/org/keycloak/connections/jpa/QuarkusJpaConnectionProviderFactory.java

```diff
@@ -240,9 +240,9 @@ public void run(KeycloakSession session) {
                      * Migrate model is executed just in case following providers are "jpa".
                      * In Map Storage, there is an assumption that migrateModel is not needed.
                      */
-                    if ("jpa".equals(Config.getProvider("realm")) &&
-                        "jpa".equals(Config.getProvider("client")) &&
-                        "jpa".equals(Config.getProvider("clientScope"))) {
+                    if (Config.getProvider("realm") != null && "jpa".equals(Config.getProvider("realm")) &&
```

Comment:
`Config.getProvider("realm") != null` effectively means `Config.getProvider("realm") == "jpa"`, as `jpa` is (currently) a default provider.

Reply:
`Config.getProvider("realm")` here returns null. There are links to code where it's resolved: 

- https://github.com/keycloak/keycloak/blob/master/quarkus/runtime/src/main/java/org/keycloak/configuration/MicroProfileConfigProvider.java#L44
- https://github.com/keycloak/keycloak/blob/master/quarkus/runtime/src/main/java/org/keycloak/configuration/MicroProfileConfigProvider.java#L64

So in order to make tests pass I've added the null check, but maybe it's not correct approach.